### PR TITLE
Proof runcountable rebase

### PR DESCRIPTION
**Pull request by VincentSe**
Kind: feature

This is a reopening of #8325, after two bad merges.

### CODE REVIEW

**Comment by VincentSe** [comment]
I implemented the smallest element in constructive epsilon and used FinFun.Bijective in the final statement of the theorem.

However it is difficult to completely replace enumeration by Bijective, as sometimes we want to have the inverse function in Sort Set.

File: CREDITS [hunk 1]
```diff
@@ -165,6 +165,7 @@ of the Coq Proof assistant during the indicated time:
   Nickolai Zeldovich (MIT 2014-2016)
   Théo Zimmermann (ORCID: https://orcid.org/0000-0002-3580-8806,
                    INRIA-PPS then IRIF, 2015-now)
+  Vincent Semeria (2018)
```
Comment (Zimmi48) [inline CREDITS]: You are supposed to insert your name in alphabetical order.

**Review by Zimmi48 — APPROVED** [review]
Doc change is trivial.

File: CREDITS (hunk 1 above)
Comment (VincentSe) [reply CREDITS]: Done

**Comment by VincentSe** [comment]
If I git rebase upstream/master, I cannot git push after, because 
To https://github.com/VincentSe/coq.git
 ! [rejected]            proofRuncountableRebase -> proofRuncountableRebase (non-fast-forward)
error: failed to push some refs to 'https://github.com/VincentSe/coq.git'
hint: Updates were rejected because the tip of your current branch is behind
hint: its remote counterpart. Integrate the remote changes (e.g.
hint: 'git pull ...') before pushing again.

I now hesitate to git pull, it messed 2 previous pull requests of mine.

**Comment by SkySkimmer** [comment]
That's normal, you need to push with force `-f` after rewriting history (which rebase does).

**Comment by VincentSe** [comment]
So what do we do about this pull request ? It has been open for about 3 months now, and I think all problems are resolved.

**Comment by VincentSe** [comment]
To prove that there is no surjection nat -> R, we could use that from any surjection nat -> A, we can (at least in classical logic) extract a bijection nat -> A (by skipping all natural numbers which images are already taken). But that theorem is more about natural numbers than real numbers : it should be proved in another file, and probably another pull request.

**Comment by herbelin** [comment]
> Currently, the proof does use both surjectivity and injectivity : when it selects the next two indices in the real interval, it assumes they correspond to real numbers that were not seen before. Hacking the proof to workaround this is about the same as proving the bijection extraction above.

OK, I see. Thanks for the explanation.

> Concerning injections, we could use the current theorem "not bijective" plus the Cantor-Bernstein theorem. I don't know if Cantor-Bernstein is implemented in Coq. Anyway it should be in another file too.

I did that long time ago. I could unbury it if needed (I used classical logic of course).

I don't have more questions, thanks a lot. So, letting @silene assigning if he has time, and, otherwise, I'll do it.

**Comment by herbelin** [comment]
Let me assign. I'll merge on next Thursday, giving some time to @silene to react if he is around.

**Comment by VincentSe** [comment]
Thanks !

**Pull request merged**